File: RayTracerModel/Camera.cs
using System.Windows.Media.Media3D;
namespace RayTracer
{
    public class Camera
    {
        public Vector3 Pos;
        public Vector3 Forward;
        public Vector3 Up;
        public Vector3 Right;

        public Camera(Vector3 pos, Vector3 lookAt)
        {
            Forward = (lookAt - pos).Normalized();
            var down = new Vector3(0, -1, 0);
            Right = Forward.CrossProduct(down).Normalized() * 1.5;
            Up = Forward.CrossProduct(Right).Normalized() * 1.5;
            Pos = pos;
        }
    }
}

File: RayTracerModelVB/Camera.vb
﻿Public Class Camera
    Public Pos As Vector3
    Public Forward As Vector3
    Public Up As Vector3
    Public Right As Vector3

    Public Sub New(ByVal pos As Vector3, ByVal lookAt As Vector3)
        Forward = (lookAt - pos).Normalized()
        Dim down = New Vector3(0, -1, 0)
        Right = Forward.CrossProduct(down).Normalized() * 1.5
        Up = Forward.CrossProduct(Right).Normalized() * 1.5
        Me.Pos = pos
    End Sub
End Class

